docs/changelog_templates.rst

```diff
@@ -0,0 +1,290 @@
+        └── static
+            └── config.cfg
+
+After running a release with Python Semantic Release, the directory structure
+of the project will now look like this:
+
+.. code-block::
+
+    example-project
+    ├── CHANGELOG.md
+    ├── src
+    │   └── example_project
+    │       ├── data
+    │       │   └── data.json
+    │       └── __init__.py
+    ├── static
+    │   └── config.cfg
+    └── templates
+        ├── CHANGELOG.md.j2
+        ├── .components
+        │   └── authors.md.j2
+        ├── .macros.j2
+        ├── src
+        │   └── example_project
+        │       └── data
+        │           └── data.json.j2
+        └── static
+            └── config.cfg
+
+Note that:
+
+* There is no top-level ``.macros`` file created, because this file is excluded
+  from the rendering process.
+* There is no top-level ``.components`` directory created, because this folder and
+  all files and folders contained within it are excluded from the rendering process.
+* To render data files into the ``src/`` folder, the path to which the template should
+  be rendered has to be created within the ``templates`` directory.
+* The ``templates/static`` folder is created at the top-level of the project, and the
+  file ``templates/static/config.cfg`` is *copied, not rendered* to the new top-level
+  ``static`` folder.
+
+You may wish to leverage this behaviour to modularise your changelog template, to
+define macros in a separate file, or to reference static data which you would like
+to avoid duplicating between your template environment and the remainder of your
+project.
+
+.. _changelog-templates-template-rendering-template-context:
+
+Template Context:
+^^^^^^^^^^^^^^^^^
+
+Alongside the rendering of a directory tree, Python Semantic Release makes information
+about the history of the project available within the templating environment in order
+for it to be used to generate Changelogs and other such documents.
+
+The history of the project is made available via the global variable ``context``. In
+Python terms, ``context`` is a `dataclass`_ with the following attributes:
+
+* ``repo_name: str``: the name of the current repository parsed from the Git url.
+* ``repo_owner: str``: the owner of the current repository parsed from the Git url.
+* ``history: ReleaseHistory``: a :py:class:`semantic_release.changelog.ReleaseHistory` instance.
+  (See :ref:`changelog-templates-template-rendering-template-context-release-history`)
+* ``filters: Tuple[Callable[..., Any], ...]``: a tuple of filters for the template environment.
+  These are added to the environment's ``filters``, and therefore there should be no need to
+  access these from the ``context`` object inside the template.
+
+Currently, two filters are defined:
+
+* ``pull_request_url: Callable[[str], str]``: given a pull request number, return a URL
+  to the pull request in the remote.
+* ``pull_request_url: Callable[[str], str]``: given a commit hash, return a URL to the
```

Comment:
Should this be `commit_hash_url` instead?

Reply:
```suggestion
* ``commit_hash_url: Callable[[str], str]``: given a commit hash, return a URL to the
```